sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```
@@ -1119,10 +1121,9 @@ abstract class ConvertUTCTimestampExprMetaBase[INPUT <: BinaryExpression](
         willNotWorkOnGpu("timezone input must be a literal string")
       case Some(timezoneShortID) =>
         if (timezoneShortID != null) {
+          // check that this timezoneID is valid
           timezoneId = GpuTimeZoneDB.getZoneId(timezoneShortID)
-          if (!GpuTimeZoneDB.isSupportedTimeZone(timezoneId)) {
-            willNotWorkOnGpu(s"Not supported timezone type $timezoneShortID.")
-          }
+          GpuTimeZoneDB.getZoneId(timezoneShortID)
```

The second call seems redundant?

---

Just confirming my understanding. `GpuTimeZoneDB.getZoneId()` can throw an exception. In that case, we will just let the exception be propagated to the caller here, so that the query will be just stopped. Because that Spark will not be able to process that query anyway. Is this right?

---

Assuming our implementation is correct, it doesn't make sense to process a nonexistent zone

---

Right, I was wondering how the exception will be handled.